engine/v2/transaction/beef/beef_service_test.go

```diff
@@ -0,0 +1,258 @@
+package beef_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/bitcoin-sv/go-sdk/chainhash"
+	sdk "github.com/bitcoin-sv/go-sdk/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/beef"
+	"github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/beef/testabilities"
+	txerrors "github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/errors"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrepareBEEF_MissingSourceTx_InSubjectTxInput(t *testing.T) {
+	// given:
+	graphBuilder := testabilities.NewTxGraphBuilder(t)
+	repository := testabilities.NewTxRepository(t, graphBuilder)
+	service := beef.NewService(repository)
+
+	var hexGen testabilities.HexGen
+	ID, err := chainhash.NewHashFromHex(hexGen.Val())
+	require.NoError(t, err, "expected to create hash from the hex gen val: %s", hexGen.Val())
+
+	// subject tx:
+	subjectTx := sdk.NewTransaction()
+	subjectTx.AddInput(&sdk.TransactionInput{SourceTXID: ID})
+
+	// when:
+	hexBEEF, err := service.PrepareBEEF(context.Background(), subjectTx)
+
+	// then:
+	require.ErrorIs(t, err, txerrors.ErrInvalidTransactionInput)
+	require.Empty(t, hexBEEF)
+}
+
+func TestPrepareBEEF_GraphTx_WithBEEFGrandparent(t *testing.T) {
+	// given:
+	graphBuilder := testabilities.NewTxGraphBuilder(t)
+	repository := testabilities.NewTxRepository(t, graphBuilder)
+	service := beef.NewService(repository)
+
+	// graph:
+	tx6 := graphBuilder.CreateMinedTx("tx6", 1)
+	tx1 := graphBuilder.CreateRawTx("tx1", testabilities.ParentTx{Tx: tx6, Vout: 0})
+	tx0 := graphBuilder.CreateRawTx("tx0", testabilities.ParentTx{Tx: tx1, Vout: 0})
+	graphBuilder.EnsureGraphIsValid()
+
+	// subject tx:
+	subjectTx := sdk.NewTransaction()
+	subjectTx.AddInput(&sdk.TransactionInput{SourceTXID: tx0.Inputs[0].SourceTXID})
+
+	// when:
+	hexBEEF, err := service.PrepareBEEF(context.Background(), subjectTx)
+
+	// then:
+	require.NoError(t, err)
+	require.NotEmpty(t, hexBEEF)
+
+	testabilities.AssertTxInputs(t, subjectTx.Inputs...)
+	testabilities.AssertTxBEEFHex(t, subjectTx)
```

Comment:
It would be good to follow the pattern of testabilities from other packages.
But it could be done in different PR.